rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -133,28 +120,129 @@
+      // create x and y axes
+      chartsEnter.append("g").attr("class", "xaxis").attr("transform", "translate(0," + ChartSettings.height + ")");
+      chartsEnter.append("g").attr("class", "yaxis");
+
+      containersEnter.merge(containers)
+        .attr("class", "chart-container " + (chartKeys.length === 1 ? 'single-chart' : 'two-charts'))
+        .each(function (index, __, charts) {
+          renderChart(d3.select(charts[index]), index);
+        })
+        .selectAll("svg")
+        .attr("width", (ChartSettings.width / chartKeys.length) + ChartSettings.left + ChartSettings.right)
+        .attr("height", ChartSettings.height + ChartSettings.top + ChartSettings.bottom)
+        .selectAll(".chart")
+        .attr("transform", "translate(" + ChartSettings.left + "," + ChartSettings.top + ")");
+
+      // Remove charts if selectedRuns reduces in size.
+      // We don't need to do this for all the inner elements, the remove will propagate.
+      containers.exit().remove();
+    }
+
+    /**
+     * Render the chart in the container.
      *
-     * @param {Object} state The current rendering state
+     * @param {Object} container D3 selector for the chart container
+     * @param {number} chartIndex The index of the chart (either 0 or 1)
      */
-    function renderD3(state) {
+    function renderChart(container, chartIndex) {
       var runs = state.runs;
       var counts = state.counts;
       var choices = state.choices;
-
-      // HACK: only show the latest run for now
-      var mostRecentRun = null;
-      if (runs.length > 0) {
-        mostRecentRun = runs[0].id;
+      var selectedRun = state.selectedRuns[chartIndex];
+
+      // select the proper option and use it to filter the runs
+      var select = container.select(".selection-row").select("select")
+        .classed("hidden", state.runs.length === 0 || state.is_open);
+      if (selectedRun === null && runs.length > 0) {
+        // The newest run should be the most recent one according to the info received from the server.
+        selectedRun = runs[0].id;
       }
 
       var histogram = choices.map(function (item) {
         return {
           answer_id: item.answer_id,
           answer_text: item.answer_text,
-          count: counts[item.answer_id][mostRecentRun] || 0
+          count: counts[item.answer_id][selectedRun] || 0
         }
       });
 
+      // D3 data join on runs to create a select list
+      var optionData = [{ id: NONE_SELECTION }].concat(runs);
+      var options = select.selectAll("option").data(optionData, function(run) {
+        return run.id;
+      });
+      options.enter()
+        .append("option")
+        .merge(options)
+        .attr("value", function(run) { return run.id; })
+        .text(function(run) {
+          if (run.id === NONE_SELECTION) {
+            if (chartIndex > 0) {
+              return 'Select previous response to compare';
+            }
+            return 'None';
```

Comment:
I was actually intending the constant to be the sentinel value for the select. I'm not sure we should use it here since this is the text representation, not the value

Reply:
Sounds good